pkg/controllers/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go

```diff
@@ -289,21 +313,33 @@ func (n *klusterletCleanupController) removeKlusterletFinalizers(ctx context.Con
 	return nil
 }
 
-func (n *klusterletCleanupController) finalClenup(ctx context.Context, config klusterletConfig) error {
-	if config.InstallMode != operatorapiv1.InstallModeHosted {
+func (n *klusterletCleanupController) deleteManifestWorks(
+	ctx context.Context, ctrlContext factory.SyncContext, config klusterletConfig) error {
+	works, err := n.controlplaneWorkClient.WorkV1().ManifestWorks(config.ClusterName).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(works.Items) == 0 {
 		return nil
 	}
 
-	if err := n.kubeClient.CoreV1().Secrets(config.AgentNamespace).Delete(
-		ctx, config.ExternalManagedClusterKubeConfigSecret, metav1.DeleteOptions{}); err != nil {
+	unavailable, err := helpers.IsClusterUnavailable(ctx, n.controlplaneClusterClient, config.ClusterName)
+	if err != nil {
 		return err
 	}
 
-	if err := n.controlplaneClusterClient.ClusterV1().ManagedClusters().Delete(
-		ctx, config.ClusterName, metav1.DeleteOptions{}); err != nil {
+	if unavailable {
+		// managed cluster is unavailable, force delete all manifestworks
+		return helpers.DeleteAllManifestWorks(ctx, n.controlplaneWorkClient, works.Items, true)
```

Comment:
Is it align with the current ACM behaviour? I remembered if the managed cluster is unavailable, the manifestwork won't be deleted forcely. correct?

Reply:
maybe plus the deletion timestamp exists in the managed cluster?

Reply:
no, if the managed cluster is unavailable, we will force delete the manifest work, when the klusterlet is deleting, if the managed cluster is  unavailable, that means we cannot do anything on the managed cluster, we can only delete the works forcely